packages/ng/libraries/core/src/lib/user/select/homonyms/user-homonyms.component.ts

```diff
@@ -0,0 +1,55 @@
+import { ChangeDetectionStrategy, Component, forwardRef, SkipSelf, Self, Optional, Inject, OnInit } from '@angular/core';
+import { ALuOptionOperator, ILuOptionOperator} from '../../../option/index';
+import { IUser } from '../../user.model';
+import { share, map, withLatestFrom, switchMap } from 'rxjs/operators';
+import { ALuUserHomonymsService, LuUserHomonymsService, ILuUserHomonymsService } from './user-homonyms.service';
+import { merge, Observable } from 'rxjs';
+
+@Component({
+	selector: 'lu-user-homonyms',
+	template: '',
+	changeDetection: ChangeDetectionStrategy.OnPush,
+	providers: [
+		{
+			provide: ALuOptionOperator,
+			useExisting: forwardRef(() => LuUserHomonymsComponent),
+			multi: true,
+		},
+		{
+			provide: ALuUserHomonymsService,
+			useClass: LuUserHomonymsService,
+		},
+	],
+})
+export class LuUserHomonymsComponent<U extends IUser = IUser> implements ILuOptionOperator<U> {
+	set inOptions$(in$: Observable<U[]>) {
+		this._outOptions$ = merge(
+			in$,
+			in$.pipe(
+				map(users => this._service.extractHomonyms(users)),
+				switchMap((homonyms: U[]) => this._service.enrichHomonyms(homonyms)),
+				withLatestFrom(in$),
+				map<[U[], U[]], U[]>(([enrichedUsers, users]) => {

```

Comment:
The race condition is not very clear. Thanks to `switchMap`, users emitted L30 should be the same as users emitted L31 (`withLatestFrom`).

It would be clearer with `map(users => [this._service.extractHomonyms(users), users]), ...` instead of `withLatestFrom(in$)`.

Reply:
indeed